lib/docs/filters/core/external_urls.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Docs
+  class ExternalUrlsFilter < Filter
+
+    def call
+      if context[:external_urls]
+
+        root = path_to_root
+
+        css('a').each do |node|
+
+          next unless anchorUrl = node['href']
+
+          # avoid links already converted to internal links
+          next if anchorUrl.match?(/\.\./)
+
+          if context[:external_urls].is_a?(Proc)
+            node['href'] = context[:external_urls].call(anchorUrl)
+            next
+          end
+
+          url = URI(anchorUrl)
+
+          context[:external_urls].each do |host, name|
+            if url.host.to_s.match?(host)
+              node['href'] = root + name + url.path.to_s + '#' + url.fragment.to_s
```

Comment:
The generated link is correct, but does not work in devdocs. Presumably a JavaScript call to `app.enableDoc(...)` is required.


https://user-images.githubusercontent.com/782446/111040714-acf78300-8434-11eb-8879-74d8412c9f5a.mov